File: Programmers/pro_12943_콜라츠추측.js
function solution(num) {
    let cnt = 0
    while (num > 1 && cnt < 500) {
        cnt += 1
        if (num % 2)
            num = num * 3 + 1
        else
            num /= 2
    }
    return cnt < 500 ? cnt : -1;
}

File: Programmers/pro_12943_콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    long long longNum = num;
    while(longNum>1 and answer++<500){
        if(longNum%2){
            longNum=longNum*3+1;
        }else{
            longNum/=2;
        }        
    }
    return answer<500 ? answer : -1;
}